### ISSUE TITLE
No icon in the extensions panel

### ISSUE THREAD

**Issue by GizMecano** [AliasIO/Wappalyzer#946]
After updating Firefox to version 40.0.3 and Wappalyzer to version 3.1.1 (according to process indicated in #930), the icon isn't displayed anymore on the extensions panel.

![wappalyzer-3 1 1-no-icon](https://cloud.githubusercontent.com/assets/852661/9561230/24aeb23a-4e3d-11e5-965b-f77bdfde9a1f.jpg)

**Comment by klonos** [comment]
Confirming this one on 42.0a1 x64 on Windows 8.1 with 3.1.1 installed through the addons update.

**Comment by AliasIO** [comment]
If this is still happening, can you please clone this repository and start Firefox using the SDK: 

https://github.com/AliasIO/Wappalyzer/wiki/Drivers#mozilla-firefox

There may be a meaningful error message in the console to help me debug the issue.

**Comment by q--** [comment]
Since the same issue applies to me, I installed Wappalyzer through the SDK. The icon wasn't showing there either. In fact, the Wappalyzer icon didn't show up at all. Yes, it was configured to be placing icons in the address bar and yes, I tried visiting a website. No luck.

Here is the output from `jpm run`:

	JPM [info] Starting jpm run on Wappalyzer
	Creating XPI
	JPM [info] XPI created at C:\Users\GEBRUI~1\AppData\Local\Temp\wappalyzer@crunchlabz.com-3.1.1.xpi (114ms)
	Created XPI at C:\Users\GEBRUI~1\AppData\Local\Temp\wappalyzer@crunchlabz.com-3.1.1.xpi
	JPM [info] Creating a new profile
	console.log: wappalyzer: [wappalyzer debug] main: install
	console.log: wappalyzer: [wappalyzer debug] w.init
	console.log: wappalyzer: [wappalyzer debug] w.driver.init
	console.error: wappalyzer: 
	  Message: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIChannel.open]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/net/url.js :: readURISync :: line 94"  data: no]
	  Stack:
		readURISync@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/net/url.js:94:16
	read@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/self.js:55:12
	w.driver.init@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/driver.js:349:23
	driver@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/wappalyzer.js:146:10
	w.init@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/wappalyzer.js:236:4
	exports.main@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/driver.js:20:3
	run@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/runner.js:151:1
	startup/</<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/runner.js:86:7
	Handler.prototype.process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:867:23
	this.PromiseWalker.walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:746:7
	this.PromiseWalker.scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:688:37

	console.error: wappalyzer: 
	  Message: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIChannel.open]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/net/url.js :: readURISync :: line 94"  data: no]
	  Stack:
		readURISync@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/net/url.js:94:16
	read@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/self.js:55:12
	w.driver.init@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/driver.js:349:23
	driver@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/wappalyzer.js:146:10
	w.init@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/wappalyzer.js:236:4
	exports.main@resource://gre/modules/commonjs/toolkit/loader.js -> resource://wappalyzer-at-crunchlabz-dot-com/driver.js:20:3
	run@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/runner.js:151:1
	startup/</<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/addon/runner.js:86:7
	Handler.prototype.process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:867:23
	this.PromiseWalker.walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:746:7
	this.PromiseWalker.scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:688:37

	console.log: wappalyzer: [wappalyzer debug] unload: shutdown

**Comment by AliasIO** [comment]
@q-- I suspect `apps.json` is missing from `src/drivers/firefox/data/`. It's not in the repository, you need to copy it there or run the `wappalyzer links` command in a Linux environment (e.g. Vagrant).

**Comment by rugk** [comment]
I can confirm this issue with Firefox 41.0.2.
![noicon](https://cloud.githubusercontent.com/assets/11966684/10556261/103ac92a-747f-11e5-8173-bcfc8ceadbf7.PNG)

**Comment by eduardodelgado** [comment]
I still have this bug with jpm 1.0.6 version. I solved it adding the next icon path in package.json:

```
    {
    
      "id": "jid1-AKuBexDY5zJsIQ@jetpack",
      "icon": "resource://jid1-akubexdy5zjsiq-at-jetpack/data/images/icon48.png",
    
    }

```
Note that the "path" is similar to the "id" field but there are not capital letters

**Comment by rugk** [comment]
@eduardodelgado If you already found a way to solve it, what about creating a PR?

**Comment by GizMecano** [comment]
No news about this (minor) problem since a while. I don't know if it can be useful, but I am experiencing [the same kind of problem](https://discussion.evernote.com/topic/96798-problem-with-the-display-in-the-extensions-panel/) with another addon recently ([Evernote Web Clipper](https://addons.mozilla.org/fr/firefox/addon/evernote-web-clipper/)).

**Comment by rugk** [comment]
Also happening here: https://github.com/bwinton/SnoozeTabs/issues/13

Also please reopen this issue. It is not fixed AFAIK.

**Comment by GizMecano** [comment]
Agreeing with @rugk : this issue seems still relevant with the last version ([3.2.4](https://addons.mozilla.org/fr/firefox/addon/wappalyzer/versions/?page=1#version-3.2.4)) released on 11.06.2016.

**Comment by rugk** [comment]
...on 2016-06-11 :wink: 

(that's why I prefer ISO formats for dates, ...)

**Comment by rugk** [comment]
Right now this is fixed AFAIK. At least I see an icon.

**Comment by GizMecano** [comment]
As @rugk has noticed some days ago, this issue seems to have been fixed with the [version 3.2.7](https://addons.mozilla.org/fr/firefox/addon/wappalyzer/versions/?page=1#version-3.2.7).
